src/peripheral/itm.rs

```diff
@@ -174,42 +186,188 @@ pub struct ITMSettings {
     pub timestamp_clk_src: TimestampClkSrc,
 }
 
+/// Possible errors on [ITM::configure].
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub enum ITMConfigurationError {
```

Comment:
perhaps `non_exhaustive`? Or are you sure these are the only errors `ITM::configure` will ever return?

Reply:
Good catch. At least one more error enum will be added in the future.